Source/Tests/LinuxSyscalls/Syscalls.h

```diff
@@ -349,6 +349,18 @@ uint64_t CloneHandler(FEXCore::Core::CpuStateFrame *Frame, FEX::HLE::clone3_args
     return flags;
   }
 
+
+template<typename T>
+static const bool HasSyscallError(T Result) {
+  // Most raw syscalls return errors using the negative range
+  // These are guaranteed to not return a valid result in the range of [-4095, -1]
+  // glibc typically wraps these results and sets errno to -Result and returns -1 from the handler
+  // In cases where FEX needs to use raw syscalls, this helper checks for this idiom
+  // !!! WARNING !!!
+  // Some syscalls may return valid results in this range, like ioctl
+  // Don't use this in those cases
+  return reinterpret_cast<uint64_t>(Result) > -4096;
```

Comment:
Is there a reason to compare against `-4096` rather than checking for negativity?

Reply:
Checking for negativity won't work since something like 0xFFFF'FFFF'FFFF'0000 is a valid result.
Could check the result being `return reinterpret_cast<uint64_t>(Result) > 0xFFFF'FFFF'FFFF'0000ULL;` If you want that to be more clear.

Reply:
Oh, of course. Using the unsigned hex constant sounds good.

Reply:
Changed to unsigned hex constant.